### PR TITLE
Implemented a toString function for the datatype base class

### PR DESCRIPTION
**Pull request by holmesie**
Implemented a toString function for the datatype base class. This will enable better output of CQL execution results for things like Excel export.

---

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1442
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A

**Reviewer 1:**

Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @mayerm94 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/datatypes/datatype.js.coffee [hunk 1]
```diff
@@ -55,3 +55,25 @@ class CQL_QDM.QDMDatatype
...
+    # Grab description from entry; if blank use classname
+    description = if @entry?.description then "#{@entry.description}\n" else "#{@constructor.name}\n"
+    # Grab start and end time, format for proper display
+    startTime = if @entry?.start_time then "START: #{moment.utc(@entry.start_time, 'X').format('MM/DD/YYYY h:mm A')}\n" else ""
+    endTime = if @entry?.end_time then "STOP: #{moment.utc(@entry.end_time, 'X').format('MM/DD/YYYY h:mm A')}\n" else ""
+    # Get code if this datatype has one
+    code = @code() if @_codes
+    codeDisplay = if code then "CODE: #{code['system']} #{code['code']}" else ""
+    # Return human readable representation of this datatype
+    "#{description}#{startTime}#{endTime}#{codeDisplay}".replace /\n$/, ''
```
Comment (losborne) [inline app/assets/javascripts/datatypes/datatype.js.coffee]: Could we just remove the `\n`s above in each of the string assignments rather than replacing them here?

Comment (losborne) [reply app/assets/javascripts/datatypes/datatype.js.coffee]: Oh I see, this only removes the last one rather than each of them.  Disregard above comment.

File: spec/javascripts/datatypes/datatype_spec.js.coffee [hunk 2]
```diff
@@ -38,3 +38,36 @@ describe "QDMDatatype", ->
...
+    expect(dataEntry.toString()).not.toContain('Diagnosis: Live Birth Newborn Born in Hospital')
+    expect(dataEntry.toString()).toContain('QDMDatatype')
+
+  it "toString includes START time when one exists", ->
+    dataEntry = new CQL_QDM.QDMDatatype(diagnosisEntry)
+    expect(dataEntry.toString()).toContain('START')
+    expect(dataEntry.toString()).toContain('08/31/2012 8:00 AM')
+
+  it "toString does not include START time when one does not exist", ->
+    dataEntry = new CQL_QDM.QDMDatatype({})
```
Comment (losborne) [inline spec/javascripts/datatypes/datatype_spec.js.coffee]: This doesn't hit the null branches of the `if @entry?` statements.  If you try to pass in `null`, the constructor blows up currently.  Switching `{}` for `null` here and adding a `?` to the `QDMDatatype` constructor fixes this for this PR, and then there is some future work that could be done to add null checks to classes that inherit from QDMDatatype, which wouldn't have to go along with this PR.

Comment (holmesie) [reply spec/javascripts/datatypes/datatype_spec.js.coffee]: I'll add that; it makes sense we should be checking for null there, even if that is a weird scenario.
